src/ngx_http_upsync_module.c

```diff
@@ -2724,33 +2737,99 @@ ngx_http_upsync_recv_handler(ngx_event_t *event)
 
 
 static ngx_int_t
-ngx_http_upsync_parse_init(void *data)
+ngx_http_upsync_consul_parse_init(void *data)
 {
-    ngx_upsync_conf_t                     *upsync_type_conf;
+    char                                  *buf;
+    size_t                                 parsed;
     ngx_http_upsync_ctx_t                 *ctx;
     ngx_http_upsync_server_t              *upsync_server = data;
 
-    upsync_type_conf = upsync_server->upscf->upsync_type_conf;
     ctx = &upsync_server->ctx;
 
-    if (upsync_type_conf->upsync_type == NGX_HTTP_UPSYNC_CONSUL
-        || upsync_type_conf->upsync_type == NGX_HTTP_UPSYNC_ETCD)
-    {
-        if (ngx_http_parser_init() == NGX_ERROR) {
-            return NGX_ERROR;
+    if (ngx_http_parser_init() == NGX_ERROR) {
+        return NGX_ERROR;
+    }
+
+    buf = (char *)ctx->recv.pos;
+
+    ctx->body.pos = ctx->body.last = NULL;
+
+    parsed = http_parser_execute(parser, &settings, buf, ngx_strlen(buf));
+    if (parsed != ngx_strlen(buf)) {
+        ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, 0,
+                      "upsync_consul_parse_init: parsed body size is wrong");
+        return NGX_ERROR;
+    }
+
+    if (ngx_strncmp(state.status, "OK", 2) == 0) {
+
+        if (ngx_strlen(state.http_body) != 0) {
+            ctx->body.pos = state.http_body;
+            ctx->body.last = state.http_body + ngx_strlen(state.http_body);
+
         }
+    }
 
-        ngx_http_parser_execute(ctx);
-        if (ctx->body.pos != ctx->body.last) {
-            *(ctx->body.last + 1) = '\0';
+    if (parser != NULL) {
+        ngx_free(parser);
+        parser = NULL;
+    }
+    
+    if (ctx->body.pos != ctx->body.last) {
+        *(ctx->body.last + 1) = '\0';
+
+    } else {
+        return NGX_ERROR;
+    }
+
+    return NGX_OK;
+}
+
```

Comment:
Two space line between functions?

Others is ok.

@CallMeFoxie , Can you sync the change to the nginx-upsync-1.9.x branch? Because I will release a new version and merge the branch into master. Thanks.